script.js

```diff
@@ -0,0 +1,57 @@
+const bookAdd = document.querySelector('.book');
+const title = document.querySelector('#title');
+const author = document.querySelector('#author');
+const btn = document.querySelector('.submit');
+
+if (!localStorage.getItem('bookInfo')) {
+  localStorage.setItem('bookInfo', JSON.stringify([]));
+}
+let collection;
+
+function saveBooks(book) {
+  localStorage.setItem('bookInfo', JSON.stringify(book));
+}
+
+/* eslint-disable no-use-before-define */
+
+function displayBookData() {
+  collection = JSON.parse(localStorage.getItem('bookInfo'));
+  updateUI();
+}
+displayBookData();
+
+btn.addEventListener('submit', (event) => {
+  event.preventDefault();
+  if (title.value && author.value) {
+    const obj = { title: title.value, author: author.value };
+    collection.push(obj);
+    saveBooks(collection);
+    displayBookData();
+    title.value = '';
+    author.value = '';
+  }
+});
+
+// Add book
+function updateUI() {
+  bookAdd.innerHTML = '';
+  collection.forEach((info, index) => {
+    const newBook = document.createElement('div');
+    newBook.classList.add('new-book');
+    const para = document.createElement('p');
+    para.textContent = `${info.title} By ${info.author}`;
+    const button = document.createElement('button');
+    button.textContent = 'Remove';
+    button.addEventListener('click', removeBook.bind(index));
+    newBook.appendChild(para);
+    newBook.appendChild(button);
+    bookAdd.appendChild(newBook);
+  });
+}
+
+// Remove book
+function removeBook() {
+  collection.splice(this, 1);
+  saveBooks(collection);
+  displayBookData();
+}
```

Comment:
Your functionality is still not working. Not only doesn't it display any book that is added to the screen, but it also doesn't store anything in the local storage other than an empty array. Kindly fix it as asked by previous reviewers and then ask for rereview.

Reply:
Hello code reviewer, in my local environment all was working ok. Can you please kindly check now if all is working ok before I submit it because am now in extra reviews :pray:

Reply:
Please find attached this [loom video](https://www.loom.com/share/fb10ad003f544eceb4f83f2510669a70) showing that my local storage is working as intended